### ISSUE TITLE
Post Create Command fails when cloning docker compose repo in container volume

### ISSUE THREAD

**Issue by daniellaera** [microsoft/vscode-remote-release#6538]
Once cloned repo in the container volume, the post create command `"postCreateCommand": "yarn install"` or `"postCreateCommand": "npm install"`

throws an error saying that impossible to find yarn.lock/package.json so impossible to install dependencies

- VSCode Version:
latest

Steps to Reproduce:

1. clone repository in container volume
2. // Use 'postCreateCommand' to run commands after the container is created.
  "postCreateCommand": "yarn install"
fails

example with npm
![dev_post](https://user-images.githubusercontent.com/8995710/161015114-15d89dee-721b-49bd-bb94-4d64590bdd18.png)

**Comment by JeremyGrieshop** [comment]
If you're asking me, I'm indeed using the Remote Containers (not Codespaces).  I had originally used these instructions for my `.devcontainer` configuration a long time ago:   https://code.visualstudio.com/remote/advancedcontainers/improve-performance#_use-a-named-volume-for-your-entire-source-tree

I used the "Clone repository into named volume container".  In previous versions of the extension, the `post` commands would operate from the current working directory of `/workspaces/{repo}`, which was correct.  Now, it's `/workspace`, so it can no longer find the root of the source tree.

**Comment by daniellaera** [comment]
same scenario as @JeremyGrieshop

**Comment by ijaouani** [comment]
I have the same issue.

My devcontainer is based on https://github.com/microsoft/vscode-dev-containers/tree/v0.209.6/containers/javascript-node-postgres.

If I open the repo using "Open a folder in Container" it works but not when using "Clone Reposository in container volume" since recent changes on vscode or extension.

I have replace my postCreateCommand with :

`"postCreateCommand": "pwd && yarn install"`

And the path is /workspace instead of /workspaces.

I am really confuse why I have both of them.

**Comment by Chuxel** [comment]
@ijaouani The path should be `/workspace` in this case since that is how the definition is set up ([see here](https://github.com/microsoft/vscode-dev-containers/blob/7809833342e5f863f8abad3fac68e9fb805615ef/containers/javascript-node-postgres/.devcontainer/devcontainer.json#L6)).   Is postCreateCommand trying to from `/workspaces/<blah>` ?

@JeremyGrieshop @daniellaera Do your devcontainer.json files have something set in `workspaceFolder`?  Does this match where the command is running?  If so, is the source code not in this location?

**Comment by ijaouani** [comment]
Thanks to @JeremyGrieshop I have find a workaround : Downgrade the version to 0.224.3 here :

![image](https://user-images.githubusercontent.com/27200712/161108773-5578c7ad-0f22-4faa-9ceb-1f02cb53e7d0.png)

**Comment by JeremyGrieshop** [comment]
> @ijaouani The path should be `/workspace` in this case since that is how the definition is set up ([see here](https://github.com/microsoft/vscode-dev-containers/blob/7809833342e5f863f8abad3fac68e9fb805615ef/containers/javascript-node-postgres/.devcontainer/devcontainer.json#L6)). Is postCreateCommand trying to from `/workspaces/<blah>` ?
> 
> @JeremyGrieshop @daniellaera Do your devcontainer.json files have something set in `workspaceFolder`? Does this match where the command is running? If so, is the source code not in this location?

Yep, I used the same sample from https://code.visualstudio.com/remote/advancedcontainers/improve-performance#_use-a-named-volume-for-your-entire-source-tree:  

`"workspaceFolder": "/workspace",`

And this is how it has been working for a long, long time.  The `/workspace` has always transformed into `/workspaces/${repo}` somewhere along the way, and I never bothered to question it.  I guess the scenario is of having a single, isolated workspace at `/workspace` vs multiple workspaces at `/workspaces`.  This process has always confused me, too.

**Comment by daniellaera** [comment]
my `"workspaceFolder": "/workspace",` 
is well configured too, like @ijaouani  and @JeremyGrieshop  I've always worked like so since long time, no issue until today

**Comment by Chuxel** [comment]
Ahh got it workspace**_s_** while the config says workspace (w/o s) - that does sound like a bug.

**Comment by JeremyGrieshop** [comment]
Semi-related (or perhaps a different issue):  This difference has always caused other problems.  The way it was setup and working previously meant that if you were to get the source locally and let the extension detect and "Reopen source in container", it would never work either, because it would assume `/workspace` instead.  

I just assumed we had to do it that one way and that one way for eternity (clone into named volume).

**Comment by Chuxel** [comment]
@chrmarti SImple repro: Open this in a container volume: https://github.com/Chuxel/empty/tree/python-postgres

On macOS you have to allow sharing `/workspaces` because of https://github.com/microsoft/vscode-remote-release/issues/5388.

Regardless, the npm install will fail - you can see its `/workspace` as the location its trying to use but the source code is not mounted there. So postCreateCommand obeys workspaceFolder but the source code mounting does not.

**Comment by ijaouani** [comment]
Just to provide more analysis on the subject :

Using docker inspect <devcontainerid> we have the following :

```
    "Mounts": [
        {
            "Type": "bind",
            "Source": "/workspaces/platform",
            "Destination": "/workspace",
            "Mode": "cached",
            "RW": true,
            "Propagation": "rprivate"
        },
        {
            "Type": "volume",
            "Name": "platform-145ecb8f78618e8a898f4d80473b5096",
            "Source": "/var/lib/docker/volumes/platform-145ecb8f78618e8a898f4d80473b5096/_data",
            "Destination": "/workspaces",
            "Driver": "local",
            "Mode": "z",
            "RW": true,
            "Propagation": ""
        },
        {
            "Type": "volume",
            "Name": "vscode",
            "Source": "/var/lib/docker/volumes/vscode/_data",
            "Destination": "/vscode",
            "Driver": "local",
            "Mode": "z",
            "RW": true,
            "Propagation": ""
        }
    ],
```

The second volume is the git in docker volume for the repo. The last one is used certainly by vscode nothing interesting here but the **first one** makes no sense for me from docker perspective.

The source of the bind is normally a local path and on windows /workspaces/platform does not exist at all. I am really curious how this bind is working and that it's currently targeting.

**Comment by JeremyGrieshop** [comment]
Seems to me that it's sort like creating a symlink at the docker mount level by re-mapping `/workspace` over top of the appropriate `/workspaces/{repo}`

**Comment by chrmarti** [comment]
Fixed in Remote-Containers v0.231.2 and v0.232.2-pre-release.

**Comment by ijaouani** [comment]
@chrmarti can you just put the link of the commit. Curious about it.
